### ISSUE TITLE
Not working on zsh on Manjaro Gnome and a patchy workaround for conda enviroments

### ISSUE THREAD

**Issue by ayan-iiitd** [andresgongora/synth-shell-prompt#15]
**Describe the bug**
When trying to run the prompt in `zsh` I get the error - 
```
/home/user/.config/synth-shell/synth-shell-prompt.sh:type:557: bad option: -t
/home/user/.config/synth-shell/synth-shell-prompt.sh:821: = not found
```
Then it defaults to the previous prompt.
Works like a charm on `bash`. 

**To Reproduce**
This is added at the end of my `.zshrc`

```
##-----------------------------------------------------
## synth-shell-prompt.sh
if [ -f /home/user/.config/synth-shell/synth-shell-prompt.sh ] && [ -n "$( echo $- | grep i )" ]; then
	source /home/user/.config/synth-shell/synth-shell-prompt.sh
fi
```

**Expected behavior**
Get the same prompt as bash.

**Something Extra**
If using `conda` instead of pure `python` to create virtual environments, changing lines 648-651`/home/user/.config/synth-shell/synth-shell-prompt.sh` from -
```
getPyenv()
{
	if [ -n "$VIRTUAL_ENV" ]; then
	    local pyenv=$(basename ${VIRTUAL_ENV})
	    echo "$pyenv"
	fi
}
```
to
```
getPyenv()
{
	local pyenv=$(basename ${CONDA_DEFAULT_ENV})
	echo "$pyenv"
}
```
works in `bash`. 

So maybe a check can be placed if a conda or python environment is on!!

**Comment by andresgongora** [comment]
Hi, thanks for the bug report. I'll try to fix it as soon as I can. I've never really tested it in ''zsh' in a VERY long time, so bugs have certainly crept in...

As for conda, it should be relatively straightforward to add an extra if statement within `getPyenv()` to check which one to use. The thing is I don't use conda myself (at least at the moment of writing) but feel free to pull request a fix if you implement it :) Otherwise, with time I'll eventually get to it ;)

**Comment by ayan-iiitd** [comment]
> As for conda, it should be relatively straightforward to add an extra if statement within `getPyenv()` to check which one to use. 

I don't exactly know if conda has a function in which you can check whether you are already in an environment or not and as of now, I am not seeing anything as such in the docs, but if you are then `local pyenv=$(basename ${CONDA_DEFAULT_ENV})` does the trick.

**Comment by ParthBibekar** [comment]
> ```
> getPyenv()
> {
> 	local pyenv=$(basename ${CONDA_DEFAULT_ENV})
> 	echo "$pyenv"
> }
> 

Changing this, gives me the following message every-time you launch the terminal or input a command

```
basename: missing operand
Try 'basename --help' for more information.
```

**Comment by andresgongora** [comment]
Sorry for the late reply. Lately I'm 100% on a different project but I should be able to resume maintenance on this repo soon :)

I don't like the idea to completely change `getPyenv()` for conda only. Maybe we should make it check `${CONDA_DEFAULT_ENV}` first (check if it is defined, then pass it to basename), and then if it's not defined, do the same with `VIRTUAL_ENV`, and if neither is set, fail elegantly. It should be easy to implement. Feel free to pull-request the cange if you implement it before I do

**Comment by andresgongora** [comment]
Please see #16 for Conda

**Comment by andresgongora** [comment]
Conda has been addressed and closed in #16 
As for zsh, we have just created a dedicated ticket with #18 , so we can address it more specifically.
Even though this is the original ticket, I'm closing it to avoid duplicity. Let's see if we can fix the zsh problem :)
